src/dungeon_range.c

```diff
@@ -0,0 +1,48 @@
+#include "global.h"
+#include "dungeon_range.h"
+
+#include "dungeon_global_data.h"
+#include "dungeon_map_access.h"
+#include "map.h"
+
+bool8 InSameRoom_2(struct Position *pos1, struct Position *pos2)
+{
+    u8 pos1RoomIndex;
+    u8 visibility = gDungeonGlobalData->visibility;
+    struct MapTile *tile1;
+    if (visibility == 0)
+    {
+        visibility = 2;
+    }
+    tile1 = GetMapTileAtPosition(pos1->x, pos1->y);
+    pos1RoomIndex = tile1->roomIndex;
+    if (pos1RoomIndex == CORRIDOR_ROOM_INDEX)
+    {
+        s32 xDiff = pos1->x - pos2->x;
+        s32 yDiff;
+        xDiff = xDiff < 0 ? -xDiff : xDiff;
+        if (xDiff > visibility)
+        {
+            return FALSE;
+        }
+
+        yDiff = pos1->y - pos2->y;
+        yDiff = yDiff < 0 ? -yDiff : yDiff;
+        if (yDiff > visibility)
+        {
+            return FALSE;
+        }
+        returnTrue:
```

Comment:
Does moving this outside of the if/else match? Seems like a tail merge..

Reply:
No, it doesn't match. The return-true label is at the end of the if block, and moving it outside the if/else puts the label at the end of the function.